beacon-chain/rpc/prysm/v1alpha1/validator/proposer_epbs.go

```diff
@@ -26,3 +30,29 @@ func (vs *Server) SubmitSignedExecutionPayloadEnvelope(ctx context.Context, env
 
 	return nil, nil
 }
+
+// computePostPayloadStateRoot computes the state root after an execution
+// payload envelope has been processed through a state transition and
+// returns it to the validator client.
+func (vs *Server) computePostPayloadStateRoot(ctx context.Context, envelope interfaces.ROExecutionPayloadEnvelope) ([]byte, error) {
+	beaconState, err := vs.StateGen.StateByRoot(ctx, envelope.BeaconBlockRoot())
+	if err != nil {
+		return nil, errors.Wrap(err, "could not retrieve beacon state")
+	}
+	beaconState = beaconState.Copy()
+	err = epbs.ProcessPayloadStateTransition(
+		ctx,
+		beaconState,
+		envelope,
+	)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not calculate post payload state root at slot %d", beaconState.Slot())
+	}
+
+	root, err := beaconState.HashTreeRoot(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not calculate post payload state root at slot %d", beaconState.Slot())
+	}
+	log.WithField("beaconStateRoot", fmt.Sprintf("%#x", root)).Debugf("Computed state root")
```

Comment:
We might want to change this log to look different from the one we have for the consensus phase. We already have a "computed state root" log when the proposer requests the beacon block. Maybe we could label this as `Computed state root at execution stage` or something similar